CONTRIBUTING.adoc

```diff
@@ -0,0 +1,259 @@
+
+# Rebase 083508e..9bcba01 onto 083508e (2 command(s))
+#
+# Commands:
+# p, pick = use commit
+# r, reword = use commit, but edit the commit message
+# e, edit = use commit, but stop for amending
+# s, squash = use commit, but meld into previous commit
+# f, fixup = like "squash", but discard this commit's log message
+# x, exec = run command (the rest of the line) using shell
+# d, drop = remove commit
+#
+# These lines can be re-ordered; they are executed from top to bottom.
+#
+# If you remove a line here THAT COMMIT WILL BE LOST.
+#
+# However, if you remove everything, the rebase will be aborted.
+#
+# Note that empty commits are commented out
+```
+
+Based on my commit messages, you can see that commit `9bcba01` fixed a mistake from my first commit.
+
+It would be nice to `squash` those changes into the first commit, so that the official history does not show my mistake..uhhh...this extra commit.
+
+To do so, edit the text to change the second commits `pick` to `squash`:
+
+```bash
+pick 083508e first commit of my really cool feature or bug fix!
+squash 9bcba01 Oops missed this one thing. This commit fixes that.
+
+# Rebase 083508e..9bcba01 onto 083508e (2 command(s))
+#
+# Commands:
+# p, pick = use commit
+# r, reword = use commit, but edit the commit message
+# e, edit = use commit, but stop for amending
+# s, squash = use commit, but meld into previous commit
+# f, fixup = like "squash", but discard this commit's log message
+# x, exec = run command (the rest of the line) using shell
+# d, drop = remove commit
+#
+# These lines can be re-ordered; they are executed from top to bottom.
+#
+# If you remove a line here THAT COMMIT WILL BE LOST.
+#
+# However, if you remove everything, the rebase will be aborted.
+#
+# Note that empty commits are commented out
+```
+
+Once you have marked the commits to be squashed and exited the edit, you will prompted to change the commit message for the new, squashed, mega commit:
+
+```bash
+# This is a combination of 2 commits.
+# The first commit's message is:
+
+first commit of my really cool feature or bug fix!
+
+# This is the 2nd commit message:
+
+Oops missed this one thing. This commit fixes that.
+
+#Please enter the commit message for your changes. Lines starting
+# with '#' will be ignored, and an empty message aborts the commit.
+#
+# Date:      Thu Oct 15 09:59:23 2015 -0600
+#
+# interactive rebase in progress; onto 083508e
+# Last commands done (2 commands done):
+#    pick 09134d5 first commit of my really cool feature or bug fix!
+#    squash 9bcba01 Oops missed this one thing. This commit fixes that.
+# No commands remaining.
+# You are currently editing a commit while rebasing branch 'mywork' on '0835    08e'.
+#
+# Changes to be committed:
+...
+```
+
+Edit the two commit messages into a single message that describes the overall change:
+
+```
+
+Once you have and exit, you will have a change to change the commit message for the new, squashed, mega commit:
```

Comment:
This seems like a duplicate of line 159, with slightly different wording. Nuke it?